Zustand/Next.js flowcharts example

CTNicholas opened the pull request:
Under popular demand!

**TODO**
- [x] Add image
- [ ] Complete https://github.com/liveblocks/liveblocks.io/pull/924

![image (4)](https://user-images.githubusercontent.com/33033422/219450890-ab3b46be-c981-4e72-bfa9-ad53933dd63b.png)

marcbouchenoire commented on examples/nextjs-flowchart/package.json:
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "nextjs-flow-chart",
```
This would need to follow the folder so `nextjs-flowchart`.

marcbouchenoire commented on examples/nextjs-flowchart/pages/_app.tsx:
```diff
@@ -0,0 +1,8 @@
+import "@/styles/globals.css";
```
We don't use TS paths in many places (anywhere even maybe?) so it'd be in favor of not using them here, especially if it's just used here and is synonymous to `./styles/globals.css`.

CTNicholas replied:
Whoops! Didn't notice this, must be the new Next.js default

marcbouchenoire reviewed (changes requested):
Nice! I think the README should mention React Flow and more importantly Zustand, I first thought it was a regular Next.js/React example instead of a Zustand one.

Even the example name might be problematic, should it be `zustand-flowchart`? To me knowing that it uses `@liveblocks/zustand` instead of `@liveblocks/react` is more important than knowing that it uses Next.js.

CTNicholas commented:
> Nice! I think the README should mention React Flow and more importantly Zustand, I first thought it was a regular Next.js/React example instead of a Zustand one.
> 
> Even the example name might be problematic, should it be `zustand-flowchart`? To me knowing that it uses `@liveblocks/zustand` instead of `@liveblocks/react` is more important than knowing that it uses Next.js.

Great points, I've updated it with your suggestions 👍 also added a few comments

marcbouchenoire commented on examples/zustand-flowchart/README.md:
```diff
@@ -0,0 +1,89 @@
+  <a href="https://codesandbox.io/s/github/liveblocks/liveblocks/tree/main/examples/nextjs-flowchart">
+    <img src="https://img.shields.io/badge/open%20in%20codesandbox-message?style=flat&logo=codesandbox&color=333&logoColor=fff" alt="Open in CodeSandbox" />
+  </a>
+  <img src="https://img.shields.io/badge/react-message?style=flat&logo=react&color=0bd&logoColor=fff" alt="React" />
+  <img src="https://img.shields.io/badge/next.js-message?style=flat&logo=next.js&color=07f&logoColor=fff" alt="Next.js" />
```
Last nitpick: we should add the Zustand badge here like other Zustand examples.

marcbouchenoire commented on examples/zustand-flowchart/tsconfig.json:
```diff
@@ -0,0 +1,24 @@
+    "jsx": "preserve",
+    "incremental": true,
+    "baseUrl": ".",
+    "paths": {
+      "@/*": ["./*"]
+    }
```
Since we're not using them we can remove this from the config.

marcbouchenoire reviewed (changes requested):
I thought it was part of the template but it looks like the README image is missing the usual border.

![](https://user-images.githubusercontent.com/6959425/220084744-001ff1a3-562e-4ca9-9665-99d9f97f0b6b.png)

CTNicholas commented:
> I thought it was part of the template but it looks like the README image is missing the usual border.
> 
> ![](https://user-images.githubusercontent.com/6959425/220084744-001ff1a3-562e-4ca9-9665-99d9f97f0b6b.png)

Good spot! Fixed

marcbouchenoire reviewed (approved):
🙌
